messenger-appUITests/messenger_appUITests.swift

```diff
@@ -1,36 +1,207 @@
 //
-//  messenger_appUITests.swift
+//  messenger_appUITests1.swift
 //  messenger-appUITests
 //
-//  Created by M Jawad Khan on 20/08/2018.
+//  Created by Jay Khan on 22/08/2018.
 //  Copyright © 2018 Jay Khan. All rights reserved.
 //
 
 import XCTest
 
-class messenger_appUITests: XCTestCase {
+class messenger_appUITests: messenger_appUITestsCase {
+    
+    // It checks the sign up form exists
+    func testSignUpFormExists() {
         
-    override func setUp() {
-        super.setUp()
+        app.buttons["Sign Up"].tap()
         
-        // Put setup code here. This method is called before the invocation of each test method in the class.
+        XCTAssert(app.buttons["Sign Up"].exists)
+        XCTAssert(app.textFields["Email"].exists)
+        XCTAssert(app.textFields["Password"].exists)
+        XCTAssert(app.buttons["OK"].exists)
+    }
+    
+    // It checks the login up form exists
+    func testLoginFormExists() {
         
-        // In UI tests it is usually best to stop immediately when a failure occurs.
-        continueAfterFailure = false
-        // UI tests must launch the application that they test. Doing this in setup will make sure it happens for each test method.
-        XCUIApplication().launch()
+        XCTAssert(app.buttons["Log In"].exists)
+        XCTAssert(app.textFields["Email"].exists)
+        XCTAssert(app.textFields["Password"].exists)
+        XCTAssert(app.buttons["OK"].exists)
+    }
 
-        // In UI tests it’s important to set the initial state - such as interface orientation - required for your tests before they run. The setUp method is a good place to do this.
+    // It checks a new user can successfully sign up
+    func testSignUpSuccess() {
+        
+        app.buttons["Sign Up"].tap()
+        
+        // generate a random number to append to email address
+        let randomNumber = arc4random()
+        let emailAdress = "test\(randomNumber)@test.com"
+        let emailTextField = app.textFields["Email"]
+        emailTextField.tap()
+        emailTextField.typeText(emailAdress)
+        
+        let passwordTextField = app.textFields["Password"]
+        passwordTextField.tap()
+        passwordTextField.typeText("password")
+        
+        app.buttons["OK"].tap()
+        
+        let toolbar = app.toolbars["Toolbar"]
+        toolbar.textViews["New Message"].tap()
+        XCTAssert(toolbar.buttons["Send"].exists)
+    }
+ 
+    // It checks a new user cannot sign up if they do not exter an email address
+    func testSignUpFailEmailAddressMissing() {
+        
+        app.buttons["Sign Up"].tap()
+        
+        let emailTextField = app.textFields["Email"]
+        emailTextField.tap()
+        
+        let passwordTextField = app.textFields["Password"]
+        passwordTextField.tap()
+        passwordTextField.typeText("password")
+        
+        app.buttons["OK"].tap()
+        
+        XCTAssert(app.buttons["OK"].exists)
```

Comment:
For now this is fine but once we figure out how we should mock the database so we don't fill it up with test accounts (we're limited to 100 users)

Reply:
yep, I did some research but could not find an obvious answer 🙏 . Ideally for Feature tests we mock or destroy the database and records and recreate for each test like Ruby/ Rails